Source/WebCore/platform/graphics/texmap/TextureMapper.cpp

```diff
@@ -1427,6 +1509,16 @@ void TextureMapper::drawTextureExternalOES(GLuint texture, OptionSet<TextureMapp
     drawTexturedQuadWithProgram(program.get(), { { texture, program->externalOESTextureLocation() } }, flags, targetRect, modelViewMatrix, opacity);
 }
 
+Ref<TextureMapperGPUBuffer> TextureMapper::acquireBufferFromPool(size_t size, TextureMapperGPUBuffer::Type type)
+{
+    size_t ceil = nextPowerOf2(size);
+    size_t floor = ceil >> 1; // half of ceil
+    size_t mid = floor + (floor >> 1); // (1.5 times floor)
+    size_t requestSize = (size <= mid) ? mid : ceil;
```

Comment:
You can use `std::countl_zero` to simplify this code.
https://godbolt.org/z/jY8xvP6eG

Reply:
`std::countl_zero` is new to me. I applied it to `nextPowerOf2`, but I'm not entirely sure if that's the specific usage you had in mind.